### PR TITLE
Adding player national team stats

### LINKED ISSUE #222: fb_player_season_stats() for national team stats
Trying to pull national team stats using this function - see code example below. Both lines return the club level stats. Let me know if there is another function/method for this

`fb_player_season_stats("https://fbref.com/en/players/abb3bb95/Luuk-de-Jong", stat_type = 'playing_time')
fb_player_season_stats("https://fbref.com/en/players/abb3bb95/Luuk-de-Jong-National-Team-Stats", stat_type = 'playing_time')`

Thanks

### PR DESCRIPTION
**Pull request by francescozonaro**
Added a new function fb_player_national_stats. 

I've moved the Country transformation (removal of flag code) from the "scraping" function(s) to the "cleaning" function. This solves a problem created by the national stats table not having a Country column.

Things to note: Dataframes returning National/AllComps stats are different. This happened sometimes before too, because sometimes LgRank is present and sometimes it isn't, but regarding National Stats, they consistently have the LgRank column and do not have the Country column.

Line 108 is needed to clean the flag code from the name of the national team too.

devtools::test() shows a couple errors with fbref functions but it doesn't seem to be something related to my changes

Possible other implementations: If preferred, I could try to keep everything in the same fb_player_season_stats function with a parameter ("all" [default], "national" [to get only national team data], "both" [would require merging and adding empty country/lgrank info in some rows).

Fixes #222

### CODE REVIEW

File: R/internals.R [hunk 1]
```diff
@@ -95,25 +93,36 @@
...
 
   stat_df <- stat_df[-remove_rows, ]
 
-  if(any(grepl("LgRank", names(stat_df)))){
-    cols_to_transform <- stat_df %>%
-      dplyr::select(-.data[["Season"]], -.data[["Squad"]], -.data[["Country"]], -.data[["Comp"]], -.data[["LgRank"]]) %>% names()
-  } else {
-    cols_to_transform <- stat_df %>%
-      dplyr::select(-.data[["Season"]], -.data[["Squad"]], -.data[["Country"]], -.data[["Comp"]]) %>% names()
+  cols_to_transform <- stat_df %>%
```
Comment (JaseZiv) [inline R/internals.R:104]: @francescozonaro Have you tested other existing functions where this cleaning code was used to ensure it doesn't cause problems with those existing functions?

**Comment by tonyelhabr** [comment]
Looking through this PR and seeing all of the similarities to `fb_player_season_stats()`, I wonder if we could either

1. refactor out all of the shared logic into its own separate function (not exposed to the user)
2. add an argument to `fb_player_season_stats()` like `national`, which would be `FALSE` by default. if the user specifies `national = TRUE`, then we'd dispatch to code specific to parsing out the national tables.

**Comment by francescozonaro** [comment]
Definetly two valid options yeah, will get on it.

**Comment by francescozonaro** [comment]
There are still some more changes to make (and definitely some more testing to do), but I've switched to the single function approach as suggested by @tonyelhabr.

@JaseZiv, regarding your question about the .clean_player_season_stats function, it didn't seem to me that there were any other usages when I checked the first time. I'll check again, though. Is there a better way to see all the usages of a certain function than the Find in Files feature?

**Comment by tonyelhabr** [comment]
> There are still some more changes to make (and definitely some more testing to do), but I've switched to the single function approach as suggested by @tonyelhabr.
> 
> @JaseZiv, regarding your question about the .clean_player_season_stats function, it didn't seem to me that there were any other usages when I checked the first time. I'll check again, though. Is there a better way to see all the usages of a certain function than the Find in Files feature?

Find in Files is probably the best thing to use. Otherwise, the tests for `fb_player_season_stats()` themselves should tell us whether some new, unexpected behavior is happening, potentially as a consequence of added code.

Speaking of tests, it would be a good idea to add at least one new test where `fb_player_season_stats(national = TRUE, ...)`.

**Comment by francescozonaro** [comment]
Thanks! I'll send you my final version tomorrow or the day after that, also updating the docs and adding necessary tests. From that of course you are free to make any change (or ask me to do it and I'll be happy to try)

**Comment by francescozonaro** [comment]
I've re-added in this commit an else condition that I had mistakenly forgot before, which was responsible to scrape the data if the table only contained domestic league data.

Tested with a bunch of single links + the following lists (all with and without national=TRUE). Everything is working properly for what I can see. I'll now update the docs and write the tests.

```
list = fb_player_urls("https://fbref.com/en/squads/5bb68ee3/2019-2020/Maidstone-United-Stats")
df <- fb_player_season_stats(list, stat_type = 'playing_time')

list = fb_player_urls("https://fbref.com/en/squads/fd962109/2012-2013/Fulham-Stats")
df <- fb_player_season_stats(list, stat_type = 'playing_time', national=TRUE)

list = fb_player_urls("https://fbref.com/en/squads/aa11fb42/Alaves-Women-Stats")
df <- fb_player_season_stats(list, stat_type = 'standard', national=TRUE)
```

Note: I also removed the following warning with the latest changes to the .clean_player_season_stats function. As before, this change should affect only the fb_player_season_stats() function, which is working.

```
Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.
ℹ Please use `all_of(var)` (or `any_of(var)`) instead of `.data[[var]]`
```

Note: The progressbar isn't properly showing while scraping less than 3 players. (First two are "skipped" in the pb drawing so it just outputs a space for what I can see). Not sure if that's the default behavior.

**Comment by francescozonaro** [comment]
Not really sure what happened with the tests here.. executing them locally I'm getting this

```
── Failed tests ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Failure (test-fbref.R:360:3): fb_league_stats() for players works
colnames(epl_player_shooting_22) (`actual`) and `expected_player_shooting_cols` (`expected`) don't have the same values.
* Only in `actual`:   "MP_Playing Time", "Starts_Playing Time", "Min_Playing Time", "Mins_Per_90_Playing Time", "Ast", "G+A", "G_minus_PK", "CrdY", "CrdR", ...
* Only in `expected`: "Mins_Per_90", "Sh_Standard", "SoT_Standard", "SoT_percent_Standard", "Sh_per_90_Standard", "SoT_per_90_Standard", "G_per_Sh_Standard", "G_per_SoT_Standard", "Dist_Standard", ...
```

but in the logs here there is this

```
══ Failed tests ════════════════════════════════════════════════════════════════
── Failure ('test-fbref.R:272:3'): fb_player_season_stats() works ──────────────
nrow(multiple_playing_time) == 0 is not FALSE

`actual`:   TRUE 
`expected`: FALSE

```

**Comment by francescozonaro** [comment]
Which is the wanted behavior for this player right here?

image
image

In the previous version (regardless all the National stuff) the output would have been this one

image

Due to the if condition here

```
expanded_table_elements <- player_page %>%
        rvest::html_nodes(".table_container") %>%
        rvest::html_nodes("table")

if (length(expanded_table_elements) == 0) {
  stat_df <- data.frame(player_name = player_name, player_url = player_url, Country = NA_character_)
} else {
  expanded_table_idx <- c()
  ...
}

```

So for example when scraping the entire Maidstone 2019-2020 squad like this

```
list = fb_player_urls("https://fbref.com/en/squads/5bb68ee3/2019-2020/Maidstone-United-Stats")
df <- fb_player_season_stats(list, stat_type = 'playing_time')
```

a row similar to the one above (with every other value at NA) can be found inside the DF. I've not touched that but I'm not sure that's the actual wanted behavior. I've noticed it because I added the following check

`stat_df <- stat_df[rowSums(stat_df != "" & !is.na(stat_df)) > 0, ]`

to remove empty rows in national stats tables

image

**Comment by francescozonaro** [comment]
I've updated the docs and added a couple tests but I'm a bit lost on why they are failing here.. I'm not having any problem locally

**Comment by JaseZiv** [comment]
> Which is the wanted behavior for this player right here?
> 
> image image
> 
> In the previous version (regardless all the National stuff) the output would have been this one
> 
> image
> 
> Due to the if condition here
> 
> ```
> expanded_table_elements <- player_page %>%
>         rvest::html_nodes(".table_container") %>%
>         rvest::html_nodes("table")
> 
> if (length(expanded_table_elements) == 0) {
>   stat_df <- data.frame(player_name = player_name, player_url = player_url, Country = NA_character_)
> } else {
>   expanded_table_idx <- c()
>   ...
> }
> ```
> 
> So for example when scraping the entire Maidstone 2019-2020 squad like this
> 
> ```
> list = fb_player_urls("https://fbref.com/en/squads/5bb68ee3/2019-2020/Maidstone-United-Stats")
> df <- fb_player_season_stats(list, stat_type = 'playing_time')
> ```
> 
> a row similar to the one above (with every other value at NA) can be found inside the DF. I've not touched that but I'm not sure that's the actual wanted behavior. I've noticed it because I added the following check
> 
> `stat_df <- stat_df[rowSums(stat_df != "" & !is.na(stat_df)) > 0, ]`
> 
> to remove empty rows in national stats tables
> 
> image

If I've read this correctly, I think we'd still want to see that player appear (with empty results), and find a different way to remove the line break separators?

**Comment by francescozonaro** [comment]
Perfect, then this should already be the current behaviour.

**Review by JaseZiv — APPROVED** [review]
Looks good to me - great stuff.

Happy with it too @tonyelhabr?

**Review by tonyelhabr — APPROVED** [review]
lgtm! thanks for this. this is massive

**Pull request merged**